test/bpf_tests/bpf_test.go

```diff
@@ -269,9 +274,21 @@ func loadAndPrepSpec(t *testing.T, elfPath string) *ebpf.CollectionSpec {
 		mspec.Pinning = ebpf.PinNone
 	}
 
-	// Detect program type mismatches
 	var progTestType ebpf.ProgramType
-	for _, prog := range spec.Programs {
+	for progName, prog := range spec.Programs {
+		switch prog.Type {
+		case ebpf.XDP, ebpf.SchedACT, ebpf.SchedCLS:
+		case ebpf.UnspecifiedProgram:
```

Comment:
What does `ebpf.UnspecifiedProgram` match? :thinking:

Reply:
Actually most programs in cilium since we tend to no start our ELF section names with `SEC("tc/` or `SEC("xdp/` so the loader has no clue what program types its dealing with. All of the tests have the mandatory program types as part of the declaration which we then use to infer the rest of the program types.